pkg/operator/operator.go

```diff
@@ -60,38 +65,37 @@ func New(config operatorconfig.Config) (*Operator, error) {
 		return nil, fmt.Errorf("failed to create operator controller: %v", err)
 	}
 
-	kubeClient, err := operatorclient.NewClient(kubeConfig)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create kube client: %v", err)
-	}
 	return &Operator{
 		manager: operatorManager,
 
 		// TODO: These are only needed for the default dns stuff, which
 		// should be refactored away.
-		client: kubeClient,
+		client: operatorManager.GetClient(),
 	}, nil
 }
 
 // Start creates the default DNS and then starts the operator
 // synchronously until a message is received on the stop channel.
 // TODO: Move the default DNS logic elsewhere.
 func (o *Operator) Start(stop <-chan struct{}) error {
-	// Periodicaly ensure the default controller exists.
+	// Periodicaly ensure the default dns exists.
 	go wait.Until(func() {
-		if err := o.ensureDefaultDNS(); err != nil {
-			logrus.Errorf("failed to ensure default dns: %v", err)
+		if !o.manager.GetCache().WaitForCacheSync(stop) {
```

Comment:
@Dan why is `GetCache()` called if `Manager` uses a non-caching client?

Reply:
See https://github.com/openshift/cluster-dns-operator/pull/116#discussion_r292221318

(p.s. you tagged the wrong dan)

Reply:
Also, the cache is actually still involved in the underlying event system — the rationale discussed in https://github.com/openshift/cluster-ingress-operator/pull/244#discussion_r290808716 applies here too. Not strictly necessary, but in terms of lifecycle it seemed sane to wait for the framework to hit something resembling a steady state before doing anything else.